### PR TITLE
Permissions: adapt to 24.2

### PR DESCRIPTION
**Pull request by fschinkel**
The id of a permission is a typed PermissionId from 24.2 on and String in earlier versions. Adjust generators accordingly.

381067

### CODE REVIEW

File: org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/java/generator/method/IScoutFieldGenerator.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.sdk.core.s.java.generator.method;
+
+import org.eclipse.scout.sdk.core.java.generator.field.IFieldGenerator;
+
+/**
+ * <h3>{@link IScoutFieldGenerator}</h3>
+ */
+public interface IScoutFieldGenerator<TYPE extends IScoutFieldGenerator<TYPE>> extends IFieldGenerator<TYPE> {
```
Comment (mvilliger) [inline org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/java/generator/method/IScoutFieldGenerator.java]: what is this class for?

File: org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java [hunk 2]
```diff
@@ -10,33 +10,82 @@
...
+    return new PermissionGenerator();
+  }
 
   @Override
   protected void setup() {
+    withPreProcessor((e, c) -> {
+      withoutField(field -> field.elementName(c).map(ID_FIELD_NAME::equals).orElse(false));
+      idRef()
+          .filter(ID_FIELD_NAME::equals)
+          .ifPresent(idRef -> withIdRef(null));
```
Comment (mvilliger) [inline org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java:40]: why is this necessary?

File: org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java [hunk 3]
```diff
@@ -10,33 +10,82 @@
...
+          b.superClause().parenthesisOpen();
+          idRef().ifPresentOrElse(
+              idRef -> {
+                // split into class and field if possible
+                // "ID" -> "ID"
+                // "a.b.c.d.SomeClass.MY_FIELD" -> "a.b.c.d.SomeClass" & "MY_FIELD"
+                var lastDotIndex = idRef.lastIndexOf(JavaTypes.C_DOT);
+                if (lastDotIndex < 0) {
+                  b.ref(idRef);
+                  return;
```
Comment (mvilliger) [inline org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java]: this is no reference, just a string to append here? right?
The field name can just be appended with append() like below?

File: org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java [hunk 4]
```diff
@@ -10,33 +10,82 @@
...
+  protected IMethodGenerator<?, ?> createConstructor() {
+    var permissionName = elementName().orElseThrow(() -> newFail("Permission name is missing"));
     return MethodGenerator.create()
         .asPublic()
         .withElementName(permissionName)
-        .withBody(b -> b.superClause().parenthesisOpen().stringLiteral(permissionName).parenthesisClose().semicolon());
+        .withBody(b -> {
+          b.superClause().parenthesisOpen();
+          idRef().ifPresentOrElse(
+              idRef -> {
```
Comment (mvilliger) [inline org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java]: extract this lambda into own function with a descriptive name?

File: org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java (hunk 2 above)
Comment (fschinkel) [reply org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java:40]: Consider the following: A generator is reused and the first run created an `ID`-constant but for the second run an external reference is passed. In this case we need to remove the `ID`-field before the second run.
